core/templates/dev/head/components/profileLink.js

```diff
@@ -0,0 +1,21 @@
+oppia.directive('profileLinkText', [function() {
+  return {
+    restrict: 'E',
+    scope: {
+      linkedName: '@'
+    },
+    templateUrl: 'components/profileLinkText',
+  };
+}]);
+
+oppia.directive('profileLinkImage', [function() { 
+  return {
+    restrict: 'E',
+    transclude: true,
+    scope: {
+      linkedName: '@',
+      linkedImage: '@'
```

Comment:
Oh, wait, sorry -- my bad. That handler only gives the current user's profile picture, but what you actually want is the profile picture of an arbitrary user. I think you'd want to expose a new handler in core/controllers/profile.py that takes a user_id, gets the user object, and returns the appropriate picture.